src/core/mesh/qgsmeshdataset.cpp

```diff
@@ -435,3 +434,180 @@ void QgsMesh3dDataBlock::setValid( bool valid )
   mIsValid = valid;
 }
 
+QgsMeshDatasetGroupTreeItem::QgsMeshDatasetGroupTreeItem() = default;
+
+QgsMeshDatasetGroupTreeItem::QgsMeshDatasetGroupTreeItem( const QString &defaultName,
+    bool isVector,
+    int index )
+  : mProviderName( defaultName )
+  , mIsVector( isVector )
+  , mDatasetGroupIndex( index )
+{
+}
+
+QgsMeshDatasetGroupTreeItem::QgsMeshDatasetGroupTreeItem( const QDomElement &itemElement, const QgsReadWriteContext &context )
+{
+  Q_UNUSED( context );
+  if ( itemElement.hasAttribute( QStringLiteral( "display-name" ) ) )
+    mUserName = itemElement.attribute( QStringLiteral( "display-name" ), mUserName );
+
+  if ( itemElement.hasAttribute( QStringLiteral( "provider-name" ) ) )
+    mProviderName = itemElement.attribute( QStringLiteral( "provider-name" ), mProviderName );
+
+  if ( itemElement.hasAttribute( QStringLiteral( "is-vector" ) ) )
+    mIsVector = itemElement.attribute( QStringLiteral( "is-vector" ) ).toInt();
+
+  if ( itemElement.hasAttribute( QStringLiteral( "dataset-index" ) ) )
+    mDatasetGroupIndex = itemElement.attribute( QStringLiteral( "dataset-index" ) ).toInt();
+
+  if ( itemElement.hasAttribute( QStringLiteral( "is-enabled" ) ) )
+    mIsEnabled = itemElement.attribute( QStringLiteral( "is-enabled" ) ).toInt();
+
+  QDomElement childElement = itemElement.firstChildElement( QStringLiteral( "mesh-dataset-group_tree-item" ) );
```

Comment:
I would suggest not to use a mix of underscores and dashes in a single element name...

Reply:
of course, typo error...